Mage.Sets/src/mage/sets/eldritchmoon/SoulSeparator.java

```diff
@@ -0,0 +1,117 @@
+import mage.target.common.TargetCardInYourGraveyard;
+
+/**
+ *
+ * @author escplan9 (Derek Monturo - dmontur1 at gmail dot com)
+ */
+public class SoulSeparator extends CardImpl {
+
+    public SoulSeparator(UUID ownerId) {
+        super(ownerId, 199, "Soul Separator", Rarity.RARE, new CardType[]{CardType.ARTIFACT}, "{3}");
+        this.expansionSetCode = "EMN";
+
+        // {5}, {T}, Sacrifice Soul Separator: Exile target creature card from your graveyard. 
+        // Put a token onto the battlefield that's a copy of that card except it's 1/1, it's a Spirit in addition to its other types, and it has flying. 
+        // Put a black Zombie creature token onto the battlefield with power equal to that card's power and toughness equal that card's toughness.
+        PutTokenOntoBattlefieldCopyTargetEffect copyEffect = new PutTokenOntoBattlefieldCopyTargetEffect(null, null, false, 1, false, false, null, 1, 1, true);
+        copyEffect.setAdditionalSubType("Spirit");
+        copyEffect.setText("Put a token onto the battlefield that's a copy of that card except it's 1/1, it's a Spirit in addition to its other types, and it has flying.");
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, copyEffect, new ManaCostsImpl("{5}"));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new SacrificeSourceCost());
+        ability.addTarget(new TargetCardInYourGraveyard(new FilterCreatureCard("creature card from your graveyard")));
+        ability.addEffect(new SoulSeparatorEffect());
+        this.addAbility(ability);
+    }
+
+    public SoulSeparator(final SoulSeparator card) {
+        super(card);
+    }
+
+    @Override
+    public SoulSeparator copy() {
+        return new SoulSeparator(this);
+    }
+}
+
+class SoulSeparatorEffect extends OneShotEffect {
+
+    public SoulSeparatorEffect() {
+        super(Outcome.PutCreatureInPlay);
+        this.staticText = "Put a black Zombie creature token onto the battlefield with power equal to that card's power and toughness equal that card's toughness.";
+    }
+
+    public SoulSeparatorEffect(final SoulSeparatorEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public SoulSeparatorEffect copy() {
+        return new SoulSeparatorEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Card creatureCard = game.getCard(this.getTargetPointer().getFirst(game, source));
+        Player controller = game.getPlayer(source.getControllerId());
+        if (creatureCard != null && controller != null) {
+            boolean result = false;
+            if (game.getState().getZone(creatureCard.getId()).equals(Zone.GRAVEYARD)) {
+                result = controller.moveCardToExileWithInfo(creatureCard, null, "", source.getSourceId(), game, Zone.GRAVEYARD, true);
+                ZombieToken token = new ZombieToken();                
+                token.power.setValue(creatureCard.getPower().getValue());
```

Comment:
I would prefer to make a new ZombieToken class that has power and toughness as input parameters.

And don't switch MageObjectImpl attributes to public without real need.